File: Imperativo/Practica_1/Ejercicio_2/Ejercicio_2.go
/**
Instituto Tecnologico de Costa Rica
Escuela Computacion 
Sede San Carlos 

Curso: Lenguages de Programacion
Profesor: Oscar Mario Viquez Acuna

Estudiante: Josue Chaves Araya - 2015094068

Ejercicio 2 de Lenguages Imperativos 
**/

package main

import (
	"fmt"
	"os"
	"strconv"
)

func draw(centerLine int){

	middle := centerLine/2
	ast := -1

	for i := middle; i > (middle*-1)-1; i-- {

		str := ""
		spaces := i

		if spaces < 0 {
			spaces *=-1	
			ast -= 2 
		}else{
			ast+=2
		}

		for j := 0; j < spaces; j++ {
			str +="  "
		}

		for j := 0; j < ast; j++ {
			str += "* "
		}

		fmt.Println(str)
	}

}


func main(){

	centerLineElemStr := os.Args[1]
	

	//Convert command line argument to Integer
	centerLineElem, err := strconv.Atoi(centerLineElemStr)
	if err != nil {
		fmt.Println("Error during conversion")
		return
	}

	if centerLineElem <=0 || centerLineElem%2 != 1 {
		fmt.Println("Input must be an odd number greater than 0")
		return
	}

	fmt.Println("Drawing figure with center line of:",centerLineElem)
	draw(centerLineElem)

}

File: Logico/Practica_1/Ejercicio_2/Ejercicio_2.pl
/*
Instituto Tecnologico de Costa Rica
Escuela Computacion 
Sede San Carlos 

Curso: Lenguages de Programacion
Profesor: Oscar Mario Viquez Acuna

Estudiante: Josue Chaves Araya - 2015094068

Ejercicio 2 de Lenguage Lógico 


Construya una función que se llame sub_conjunto. Esta función recibe dos listas y debe 
retornar True cuando el primer argumento es subconjunto completo del segundo y #f en 
caso contrario. 

Por ejemplo:  
  sub_conjunto([],[1,2,3,4,5]). 
  True 
  sub_conjunto([1,2,3],[1,2,3,4,5]). 
  True 
  sub_conjunto([1,2,6],[1,2,3,4,5]). 
  False 
*/


sub_conjunto([],_).          % Un conjunto vacío es subconjunto de cualquier conjunto 
sub_conjunto([H|T], R) :-    % Cada elemento de [H|T] es subconjunto de R sí y sólo sí: 
  member(H, R),             % H es miembro de R
  sub_conjunto(T, R),!.      % y el resto de los elementos de T también son subconjunto de R.  
